crates/re_data_store/src/entity_tree.rs

```diff
@@ -73,6 +58,54 @@ impl StoreSubscriber for EntityTree {
     }
 }
 
+/// Information about this specific entity (excluding children).
+#[derive(Default)]
+pub struct EntityInfo {
+    /// Book-keeping around whether we should clear fields when data is added.
+    clears: BTreeMap<RowId, TimePoint>,
+
+    /// Flat time histograms for each component of this [`EntityTree`].
+    ///
+    /// Keeps track of the _number of times a component is logged_ per time per timeline, only for
+    /// this specific [`EntityTree`].
+    /// A component logged twice at the same timestamp is counted twice.
+    ///
+    /// ⚠ Auto-generated instance keys are _not_ accounted for. ⚠
+    pub components: BTreeMap<ComponentName, TimeHistogramPerTimeline>,
+}
+
+/// Info about stuff at a given [`EntityPath`], including all of its children, recursively.
+#[derive(Default)]
+pub struct SubtreeInfo {
+    /// Book-keeping around whether we should clear recursively when data is added.
+    clears: BTreeMap<RowId, TimePoint>,
+
+    /// Recursive time histogram for this [`EntityTree`].
+    ///
+    /// Keeps track of the _number of components logged_ per time per timeline, recursively across
+    /// all of the [`EntityTree`]'s children.
+    /// A component logged twice at the same timestamp is counted twice.
+    ///
+    /// ⚠ Auto-generated instance keys are _not_ accounted for. ⚠
+    pub time_histogram: TimeHistogramPerTimeline,
+}
+
+impl SubtreeInfo {
+    /// Assumes the event has been filtered to be part of this subtree.
```

Comment:
It there a debug-assert that could validate this assumption?

Reply:
Not without duplicating the `EntityPath`, but this is a private function, and only called from one place, which validates it